### ISSUE TITLE
Ruby 2.2 Support

### ISSUE THREAD

**Issue by esb** [rubyjedi/soap4r#12]
+1 Ruby 2.2 support

Glad to find someone still supporting soap4r! I needed to move on from the spox implementation, so I was very happy to find your contribution.

Like you, I am interested in keeping soap4r alive and am looking forward to Ruby 2.2 support.

**Comment by rubyjedi** [comment]
Aloha, thanks for the kind words and show of support for Ruby 2.2. 

I had a pull-request a few months back that gets most of this working on Ruby 2.2 with Nokogiri. As I recall, some of the other parsers weren't quite stable, and put it on a deferred task item to spend some time reworking the tests. Something really funky going on in there -- either a timing or race-condition type of issue related to legacy WEBrick and HTTPClient -- and I'm hoping that adding support for Curb will be a useful way to side-step those race conditions going forward.

Now that 2.2 is much more mainstream, I will probably spend more time this, and hopefully roll in support for Curb to get a noticable speed boost.

**Comment by stuarthannig** [comment]
+1

**Comment by detournemint** [comment]
👍

**Comment by oaguy1** [comment]
:+1:

**Comment by paopay** [comment]
+1

**Comment by rubyjedi** [comment]
Good news, all -- most of what is in Soap4R-ng is already functional with Ruby 2.2. 

The majority of 2.2-related issues with the regression tests were from the tests that used the WEBrick-based SOAP Server. It wasn't releasing the socket quickly enough before the next test started. I have added a small bit of throttling code to work around that issue, but otherwise the current Gem released a year ago is still valid and useful for Ruby 2.2.

With Ruby 2.3, it looks like there are some Ruby-Marshaling issues that need to be addressed, and I'll get to that in the upcoming weeks. Closing this issue, as Soap4R-ng on Ruby 2.2 is testing as fully-functional with the throttling code in place.
